src/m365/spo/commands/site/site-apppermission-remove.ts

```diff
@@ -0,0 +1,199 @@
+    let filterProperty: string = 'displayName';
+    let filterValue: string = args.options.appDisplayName as string;
+
+    if (args.options.appId) {
+      filterProperty = 'id';
+      filterValue = args.options.appId;
+    }
+
+    return permissions.filter((p: SitePermission) => {
+      return p.grantedToIdentities.some(({ application }: SitePermissionIdentitySet) =>
+        (application as any)[filterProperty] === filterValue);
+    });
+  }
+
+  private getPermissionIds(args: CommandArgs): Promise<string[]> {
+    if (args.options.permissionId) {
+      return Promise.resolve([args.options.permissionId!]);
+    }
+
+    return this.getPermissions()
+      .then((res: { value: SitePermission[] }) => {
+        let permissions: SitePermission[] = res.value;
+
+        if (args.options.appId || args.options.appDisplayName) {
+          permissions = this.getFilteredPermissions(args, res.value);
+        }
+
+        return Promise.resolve(permissions.map(x => x.id));
+      });
+  }
+
+  private removePermissions(permissionId: string): Promise<void> {
+    const spRequestOptions: any = {
+      url: `${this.resource}/v1.0/sites/${this.siteId}/permissions/${permissionId}`,
+      headers: {
+        'accept': 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request.delete(spRequestOptions);
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const removeSiteAppPermission: () => void = (): void => {
+      this
+        .getSpoSiteId(args)
+        .then((siteId: string): Promise<string[]> => {
+          this.siteId = siteId;
+          return this.getPermissionIds(args);
+        })
+        .then((permissionIdsToRemove: string[]): Promise<void[]> => {
+          const tasks: Promise<void>[] = [];
+
+          for (const permissionId of permissionIdsToRemove) {
+            tasks.push(this.removePermissions(permissionId));
+          }
+
+          return Promise.all(tasks);
+        })
+        .then((res: any): void => {
+          logger.log(res);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    };
+
+    if (args.options.confirm) {
+      removeSiteAppPermission();
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the apppermission from a site ${args.options.siteUrl}?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          removeSiteAppPermission();
+        }
+      });
+    }
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-u, --siteUrl <siteUrl>'
+      },
+      {
+        option: '-p, --permissionId [permissionId]'
+      },
+      {
+        option: '-i, --appId [appId]'
```

Comment:
This doesn't match the spec. We agreed to use `-i` for permissionId and no short for `appId`. https://github.com/pnp/cli-microsoft365/issues/2209#issuecomment-786657674